README.md

```diff
@@ -286,52 +292,169 @@ type Action = {
   message: {
     body: {
       active: boolean;
-      type: 'like' | 'follow'
+      type: "like" | "follow";
       targetUri: FarcasterURI;
-      schema: 'farcaster.xyz/schemas/v1/action';
-    }; 
+      schema: "farcaster.xyz/schemas/v1/action";
+    };
     account: number;
     timestamp: number;
   };
-}
+};
 ```
 
 #### Message Validation
+
 1. `schema` must be known.
 2. `active`, `type` must be present and match types.
 3. `targetUri` must be a valid FarcasterURI and must not reference this message.
 
 #### Set Construction
 
-Actions are managed with an [LWW-Element-Set CRDT](https://en.wikipedia.org/wiki/Conflict-free_replicated_data_type#LWW-Element-Set_(Last-Write-Wins-Element-Set)) which guarantees strong eventual consistency. Conceptually, there is a single **set** that stores all messages and conflicts are resolved by timestamp and lexicographical hash order. An addition is performed by constructing an `Action` message `a` where `active` is true, while a remove is performed by setting `active` to false. In both cases, the logic for merging the message into the set is: 
+Actions are managed with an [LWW-Element-Set CRDT](<https://en.wikipedia.org/wiki/Conflict-free_replicated_data_type#LWW-Element-Set_(Last-Write-Wins-Element-Set)>) which guarantees strong eventual consistency. Conceptually, there is a single **set** that stores all messages and conflicts are resolved by timestamp and lexicographical hash order. An addition is performed by constructing an `Action` message `a` where `active` is true, while a remove is performed by setting `active` to false. In both cases, the logic for merging the message into the set is:
 
 1. If there is an action `x` in the set with the same values for `type`, `targetUri` and `account` as the incoming action `y`:
    - If `x > y`, discard `y`
    - If `x < y`, delete `x` and add `y` into the set
-2. Otherwise, add `y` into the set. 
-
+2. Otherwise, add `y` into the set.
 
 ## 4.3 Verifications
-*This section is still under development and will cover a CRDT for verified data types, including proof of ownership of an address or a specific NFTs*
+
+Verifications are bi-directional proofs of ownership between Farcaster accounts and external identities. Initially, Ethereum addresses are the only supported external identities, though other blockchains can be added later.
+
+Verifications must be opted in to by both the external identity (i.e. Ethereum address) and the Farcaster account. The bi-directional verification is implemented in two steps:
```

Comment:
Unlike `Actions`, `Verifications` are more likely to have sub-types with different schemas similar to `Casts`. I think it might be worth breaking out a separate `4.3.1 ECDSA EIP-191 Verification` (or some friendlier title) so that the reader can understand the difference between the general verification idea (bi-directional proofs) and the specific implementation (bi-directional signatures from Ethereum.

Reply:
It is also work expressing that this proof type is self-authenticating

Reply:
I re-organized the section in this way:
- Generic info about verifications
- add and remove message types
- Generic message validation
- Generic set construction and pseudocode
- Custom info about the EIP 191 0x45 verification type